contracts/Marketplace1155.sol

```diff
@@ -0,0 +1,161 @@
+
+contract marketPlace1155 is ReentrancyGuard, ERC1155Receiver {
+    using Counters for Counters.Counter;
+    Counters.Counter private _itemIds;
+    Counters.Counter private _itemsSold;
+
+    address public owner;
+
+    constructor() {
+        owner = msg.sender;
+    }
+
+    struct MarketItem {
+        uint256 itemId;
+        address nftContract;
+        uint256 tokenIds; // togli s finale
+        address payable seller;
+        address payable owner;
+        uint256 price;
+        bool sold; //serve?
+    }
+
+    mapping(uint256 => MarketItem) private idToMarketItem; // a cosa mi serve la mappa. Sto rimappando i dati di 1155. Se non aggiunge informazione a cosa serve?
+
+    event MarketItemCreated(
+        uint256 indexed itemId,
+        address indexed nftContract,
+        uint256 indexed tokenIds,
+        address seller,
+        address owner,
+        uint256 price,
+        bool sold
+    );
+    // serve il bool sold? quando lo creo puo' essere gia' venduto?
+
+    event MarketItemSold(uint256 indexed itemId, address owner);
+
+    function onERC1155Received(
+        address,
+        address,
+        uint256,
+        uint256,
+        bytes memory
+    ) public virtual override returns (bytes4) {
+        return this.onERC1155Received.selector;
+    }
+
+    function onERC1155BatchReceived(
+        address,
+        address,
+        uint256[] memory,
+        uint256[] memory,
+        bytes memory
+    ) public virtual override returns (bytes4) {
+        return this.onERC1155BatchReceived.selector;
+    }
+
+    function createMarketItem(
+        address nftContract,
+        uint256[] memory tokenIds,
+        uint256 price,
+        uint256[] memory amounts // se e' un nft amounts non dovrebbe essere sempre 1? siccome immagino siano tracked dai tokenIds questo mi sembra superfluo
+    ) public payable nonReentrant {
+        require(price > 0, "Price must be greater than 0"); // Non vogliamo controllare che tokenIds sia lungo quanto amounts?
+
+        IERC1155(nftContract).safeBatchTransferFrom(
+            msg.sender,
+            address(this),
+            tokenIds,
+            amounts,
+            ""
+        );
+
+        for (uint256 i = 0; i <= (tokenIds.length - 1); i++) {
+            //davide: possiamo usare solo il < e togliere il -1?
+            _itemIds.increment();
+            uint256 itemId = _itemIds.current();
+
+            idToMarketItem[itemId] = MarketItem(
+                itemId,
+                nftContract,
+                tokenIds[i],
+                payable(msg.sender),
+                payable(address(0)),
+                price,
+                false
+            );
+
+            emit MarketItemCreated(
+                itemId,
+                nftContract,
+                tokenIds[i],
+                msg.sender,
+                address(0),
+                price,
+                false
+            );
```

Comment:
Maybe we can consider to emit a single event instead of N separate ones. Of course this can be done if tokenIds are specified within a specific range i.e. create Market item for tokenIds [1,2,3,4]